extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/caffeine/CaffeineCacheImpl.java

```diff
@@ -293,4 +309,38 @@ private <T> T cast(Object value) {
                     "An existing cached value type does not match the type returned by the value loading function", e);
         }
     }
+
+    @SuppressWarnings("unchecked")
+    private <K, V> CompletionStage<V> computeWithStats(K key, Function<K, Uni<V>> valueLoader) {
+        return (CompletionStage<V>) cache.asMap().compute(key,
+                new BiFunction<Object, CompletableFuture<Object>, CompletableFuture<Object>>() {
+                    @Override
+                    public CompletableFuture<Object> apply(Object key, CompletableFuture<Object> value) {
+                        if (value == null) {
+                            statsCounter.recordMisses(1);
+                            return (CompletableFuture<Object>) valueLoader.apply((K) key)
+                                    .map(i -> NullValueConverter.toCacheValue(i))
+                                    .subscribeAsCompletionStage();
+                        } else {
+                            LOGGER.tracef("Key [%s] found in cache [%s]", key, cacheInfo.name);
+                            statsCounter.recordHits(1);
+                            return value;
+                        }
+                    }
+                });
+    }
+
+    @SuppressWarnings("unchecked")
+    private <K, V> CompletionStage<V> computeWithoutStats(K key, Function<K, Uni<V>> valueLoader) {
+        return (CompletionStage<V>) cache.asMap().computeIfAbsent(key,
+                new Function<Object, CompletableFuture<Object>>() {
+                    @Override
+                    public CompletableFuture<Object> apply(Object key) {
+                        return valueLoader.apply((K) key)
+                                .map(i -> NullValueConverter.toCacheValue(i))
```

Comment:
`null` values are replaced with `NullValue.INSTANCE` when stored here (because Caffeine does not support `null` values) but they are never replaced back with `null` before being returned by this method. This is also true for `CaffeineCacheImpl#computeWithStats` so the best place to deal with that is probably in the `CaffeineCacheImpl#getAsync` method.